generators/chipyard/src/main/scala/ChipTop.scala

```diff
@@ -24,15 +27,16 @@ case object BuildSystem extends Field[Parameters => LazyModule]((p: Parameters)
  */
 
 class ChipTop(implicit p: Parameters) extends LazyModule with BindingScope
-    with HasTestHarnessFunctions with HasIOBinders {
+    with HasTestHarnessFunctions with HasReferenceClockFreq with HasIOBinders {
   // The system module specified by BuildSystem
   lazy val lazySystem = LazyModule(p(BuildSystem)(p)).suggestName("system")
 
-  // The implicitClockSinkNode provides the implicit clock and reset for the System
+  // The implicitClockSinkNode provides the implicit clock and reset for the system (connected by clocking scheme)
   val implicitClockSinkNode = ClockSinkNode(Seq(ClockSinkParameters(name = Some("implicit_clock"))))
 
   // Generate Clocks and Reset
-  p(ClockingSchemeKey)(this)
+  val mvRefClkFreq = p(ClockingSchemeKey)(this)
```

Comment:
Why do we even need this val? I think it should just be inlined below.

Reply:
Leaving this out breaks things. I think since the `def` is like a `lazy val` the clocking scheme function will get called in the  harness (non-lazy domain) causing a bunch of diplomacy errors in the `ChipTop`. So you 1st need to have the function called, then you can reference it with a `def`. Scala'y people can probably tell you the exact issue here.